src/utils/hooks/useGetFullReviewStructure.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState, useEffect } from 'react'
+import { FullStructure } from '../types'
+import {
+  ApplicationResponse,
+  Review,
+  ReviewAssignment,
+  ReviewResponse,
+  useGetReviewNewQuery,
+} from '../generated/graphql'
+import addEvaluatedResponsesToStructure from '../helpers/structure/addEvaluatedResponsesToStructure'
+import { useUserState } from '../../contexts/UserState'
+import addOwnedReviewResponse from '../helpers/structure/addOwnedReviewResponse'
+import addElementsById from '../helpers/structure/addElementsById'
+
+interface useGetFullApplicationStructureProps {
+  structure: FullStructure
+  reviewAssignment: ReviewAssignment
+}
+
+const useGetFullReviewStructure = ({
+  structure,
+  reviewAssignment,
+}: useGetFullApplicationStructureProps) => {
+  const [fullStructure, setFullStructure] = useState<FullStructure>()
+  const [isError, setIsError] = useState(false)
+
+  const {
+    userState: { currentUser },
+  } = useUserState()
+
+  const { data, error } = useGetReviewNewQuery({
+    variables: {
+      reviewAssignmentId: reviewAssignment.id,
+      userId: currentUser?.userId || 0,
+    },
+  })
+
+  useEffect(() => {
+    if (error) {
+      setIsError(true)
+      return
+    }
+
+    if (!data) return
+
+    addEvaluatedResponsesToStructure({
+      structure,
+      applicationResponses: data?.reviewAssignment?.application?.applicationResponses
+        ?.nodes as ApplicationResponse[],
+      currentUser,
+      evaluationOptions: {
+        isEditable: false,
+        isVisible: true,
+        isRequired: false,
+        isValid: false,
+      },
+    }).then((evaluatedStructure: FullStructure) => {
+      let newStructure: FullStructure = addElementsById(evaluatedStructure)
+
+      // here we add reviewless review repsonses
+      // generate indications of reviewless review response (numbe of non conform etc.)
+
+      const review = data?.reviewAssignment?.reviews?.nodes[0] as Review
```

Comment:
Good to add a comment here why is getting the first review. Also good to check (and log error) in case there is more than one review associated to one reviewAssignment?

Reply:
I've added a comment. There are other places where we have a relationship that looks like one to many, but is really one to one, like this one:
https://github.com/openmsupply/application-manager-web-app/blob/a3961e915dccac92e82731ee67399e57ea4999ee/src/utils/hooks/useLoadApplicationNEW.ts#L100

9e6db37b5002f55a4d9a2238e40f9f298d56c885

Reply:
Yep. I know, that's the expected case, but sometimes (for bugs) that rule might be broken, that's why adding the check before it like for this example, the previous 2 lines are: 
https://github.com/openmsupply/application-manager-web-app/blob/a3961e915dccac92e82731ee67399e57ea4999ee/src/utils/hooks/useLoadApplicationNEW.ts#L98-L99

Reply:
Cools, the log was added, and updated cause it wasn't accurate: 9dfd303d37e3b41e56cf11d32ec9de3f4897cee4

Reply:
Oopsy I have missed that too!